spec/unit/provider_spec.rb

```diff
@@ -412,6 +437,151 @@ def existing_command
       expect(child.specificity).to be > parent.specificity
     end
 
+    # Key: spec has 4 required and 1 optional part:
```

Comment:
With this giant matrix of tests, can we eliminate some of the hand-rolled variants in this file?